src/qt/bitcoingui.cpp

```diff
@@ -1494,7 +1497,9 @@ void BitcoinGUI::setAdditionalDataSyncProgress(double nSyncProgress)
 
     // If masternodeSync->Reset() has been called make sure status bar shows the correct information.
     if (nSyncProgress == -1) {
+        /*
         setNumBlocks(m_node.getNumBlocks(), QDateTime::fromTime_t(m_node.getLastBlockTime()), QString::fromStdString(m_node.getLastBlockHash()), m_node.getVerificationProgress(), false);
+        */
```

Comment:
18152: ?

Reply:
see commit description https://github.com/dashpay/dash/pull/5802/commits/e76f696bb26cae94bd167fe722f0b8cf12d5be73

Reply:
>2. Clarification for setAdditionalDataSyncProgres:
The call of masternodeSync.Reset() can happen only if chain is destroyed,
it happens if application is closed. We don't need to update num of blocks
in this situation, as well we don't need to start spinner

This is not true. We force sync reset in a couple of cases in `CMasternodeSync` itself and via `mnsync reset` RPC

Reply:
```
src/net.cpp:        ::masternodeSync->Reset();
src/net.cpp:    ::masternodeSync->Reset();
src/qt/bitcoingui.cpp:    // If masternodeSync->Reset() has been called make sure status bar shows the correct information.
src/test/util/setup_common.cpp:    ::masternodeSync.reset();
src/rpc/misc.cpp:        ::masternodeSync->Reset(true);
src/init.cpp:    ::masternodeSync.reset();
```
ah I see now, reset() and Reset() are different things. 

Could you suggest a fix for case of Reset()?

Reply:
I guess we had `setNumBlocks` here and in (1) because we wanted up to date info so it should be `POST_INIT` (i.e. disable throttling) in both cases I think.

Reply:
hmm... but `POST_INIT` enables app nap which we probably do not want.. maybe we could add a new one and use it instead?

Reply:
I would not agree about `fNetworkBecameActive || fNetworkBecameInactive` because connection can "re-connect" during to initial sync -> and it doesn't mean that chain transitioned to status "POST_INIT", we should not set this status in this case.

But I am not sure about masternodeSync::reset() case though... Can it happen any time or only on POST_INIT stage?

Reply:
Actually, I looked to usages of `masternodeSync::Reset()`:

once it is called inside `CConnman::SetNetworkActive` - so it means that it can be called if network become active/inactive... Rpc `mnsync` can be called before sync is also finish. 

So, I'd prefer to keep it as is in https://github.com/dashpay/dash/commit/e76f696bb26cae94bd167fe722f0b8cf12d5be73 now

Reply:
The only thing these sync statuses change in `setNumBlocks` is app nap state switch and we definitely do not want app nap to be enabled in both cases. `POST_INIT` is probably not a safe one then, should pick any other state, doesn't really matter which one.

`::masternodeSync.reset()` can only happen on shutdown and anything would work there imo, I would just ignore this case.

>So, I'd prefer to keep it as is in https://github.com/dashpay/dash/commit/e76f696bb26cae94bd167fe722f0b8cf12d5be73 now

I disagree, we should re-enable `setNumBlocks` and the spnner.

Reply:
can we maybe drop 18152 from this PR and do it in a separate one or will it cause some merge conflicts?

Reply:
here's it: https://github.com/dashpay/dash/pull/5804

no conflicts, but 18152 is needed for several further backports from v21/v22 (no PRs for them yet)

Reply:
> EDIT: just tried it myself, looks like it can be dropped safely

yes, I slightly tested before PR is created: run app, stop app before sync is over; restart and wait until sync is over; call re-index; also checked spinner appearance after connection to peers are established.

I didn't check all cases, but looks like main-flow works correctly with https://github.com/dashpay/dash/commit/e76f696bb26cae94bd167fe722f0b8cf12d5be73